src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/TypeLoaderExceptionHelper.cs

```diff
@@ -92,6 +98,10 @@ private static string GetFormatString(ExceptionStringID id)
                     return SR.IO_FileNotFound_FileName;
                 case ExceptionStringID.BadImageFormatGeneric:
                     return SR.Arg_BadImageFormatException;
+                case ExceptionStringID.MarshalDirectiveGeneric:
+                    return SR.Arg_BadImageFormatException;
```

Comment:
This should use more appropriate message like `SR.Arg_MarshalDirectiveException` instead of `Arg_BadImageFormatException`.